### PR TITLE
[AAE-19408] Add new logic to handle service account client

### LINKED ISSUE #4568: Add new logic to handle service account client in Keycloak
https://alfresco.atlassian.net/browse/AAE-19408

### PR DESCRIPTION
**Pull request by mavotto**
this pr fix https://github.com/Activiti/Activiti/issues/4568

### CODE REVIEW

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/client/KeycloakClient.java [hunk 1]
```diff
@@ -238,4 +240,22 @@ void updateClient(
...
+    @RequestMapping(method = RequestMethod.POST, value = "/clients/{id}/client-secret")
+    @Headers("Content-Type: application/json")
+    KeycloakCredentialRepresentation createClientSecretById(
+        @RequestBody KeycloakCredentialRequestRepresentation requestRepresentation,
+        @PathVariable("id") String id
+    );
+
+    @RequestMapping(method = RequestMethod.GET, value = "/clients/{id}/client-secret")
+    @Headers("Content-Type: application/json")
+    KeycloakCredentialRepresentation getClientSecretById(@PathVariable("id") String id);
```
Comment (pmartinezga) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/client/KeycloakClient.java:253]: Is this method needed? It looks strange to me that we publish an endpoint where you can retrieve the client secret

Comment (pmartinezga) [reply activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/client/KeycloakClient.java:253]: OK I see it exists in the Keycloak API

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java [hunk 2]
```diff
@@ -159,13 +161,14 @@ public KeycloakClient keycloakClient(
...
             clientRegistration
         );
-        KeycloakClient keycloakClient = Feign
+        return Feign
             .builder()
             .contract(new SpringMvcContract())
             .encoder(new SpringEncoder(messageConverters))
             .decoder(new SpringDecoder(messageConverters, customizers))
             .requestInterceptor(clientCredentialsAuthRequestInterceptor)
+            .logLevel(Logger.Level.FULL)
```
Comment (pmartinezga) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java]: Are we sure we want to put the level to FULL in production?

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/client/KeycloakClient.java [hunk 3]
```diff
@@ -238,4 +240,22 @@ void updateClient(
         @PathVariable("id") String id,
         @RequestBody KeycloakClientRepresentation keycloakClientRepresentation
     );
+
+    @RequestMapping(method = RequestMethod.POST, value = "/clients/{id}/client-secret")
+    @Headers("Content-Type: application/json")
+    KeycloakCredentialRepresentation createClientSecretById(
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/client/KeycloakClient.java]:
```suggestion
    KeycloakCredentialRepresentation generateNewClientSecret(
```

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java (hunk 2 above)
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java]: can we use a property for it. Thanks

**Review by jesty — CHANGES_REQUESTED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java (hunk 2 above)
Comment (mavotto) [reply activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/config/ActivitiKeycloakAutoConfiguration.java]: I forgot to remove these properties on the feign

**Review by jesty — APPROVED** [review]
LGTM

**Pull request merged**